src/main/java/BattleshipModel.java

```diff
@@ -96,12 +99,28 @@ public void AIFire(Board target)
             {
                 checkLeft = false;
             }
-            if(right.getAcross() == curr.getAcross() && right.getDown() == curr.getDown())
-            {
+            if(right.getAcross() == curr.getAcross() && right.getDown() == curr.getDown()) {
                 checkRight = false;
             }
 
-            }
+        }
+
+        if(checkUp == true){
```

Comment:
This if, else if, else should check to see if any of the 4 positions can be fired on. I think this should work because if none are fired on, the random easyAIFire function will be called to will get a new hit and then that will become the previous hit. Some testing should be done on this.